rust/ql/test/query-tests/unusedentities/unreachable.rs

```diff
@@ -1,12 +1,20 @@
-//fn cond() -> bool;
-//fn get_a_number() -> i32;
-//fn maybe_get_a_number() -> Option<i32>;
+pub fn cond() -> bool {
+    true
+}
+
+fn get_a_number() -> i32 {
+    42
+}
+
+fn maybe_get_a_number() -> Option<i32> {
+    None
+}
```

Comment:
Implementing these is nice because it makes the code buildable, but these implementations don't really represent what these functions are supposed to be.  As dataflow / control flow improves, there's a risk we eventually start to notice that `if cond()` always holds, for example.

How about (untested):
```suggestion
pub fn cond() -> bool {
    get_a_number() == 1
}

fn get_a_number() -> i32 {
    maybe_get_a_number().unwrap_or(0)
}

fn maybe_get_a_number() -> Option<i32> {
     std::env::args().nth(1).parse::<i32>()
}
```

Reply:
Done (modified slightly to make it compile).